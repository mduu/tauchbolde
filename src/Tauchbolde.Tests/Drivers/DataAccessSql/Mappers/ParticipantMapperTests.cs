using System;
using ApprovalTests;
using ApprovalTests.Reporters;
using FluentAssertions;
using Newtonsoft.Json;
using Tauchbolde.Domain.Types;
using Tauchbolde.Driver.DataAccessSql.DataEntities;
using Tauchbolde.Driver.DataAccessSql.Mappers;
using Xunit;

namespace Tauchbolde.Tests.Drivers.DataAccessSql.Mappers
{
    [UseReporter(typeof(DiffReporter))]
    public class ParticipantMapperTests
    {
        [Fact]
        public void Map()
        {
            var dataObject = new ParticipantData
            {
                Id = new Guid("BB8E0FCE-974A-4877-80A9-5E4CA315E166"),
                Status = ParticipantStatus.Tentative,
                Note = "The Note",
                CountPeople = 2,
                EventId = new Guid("C2847D0C-CAFA-4ED8-9496-10039AA6A6F5"),
                ParticipatingDiverId = new Guid("1549F94E-E376-4AC3-9591-C6938A1C9421"),
                BuddyTeamName = "Team 1",
            };

            var domainObject = dataObject.MapTo();
            var newDataObject = domainObject.MapTo();
            
            newDataObject.Should().BeEquivalentTo(dataObject);
            Approvals.VerifyJson(JsonConvert.SerializeObject(
                new
                {
                    dataObject,
                    domainObject,
                    newDataObject
                }));
        }
    }
}