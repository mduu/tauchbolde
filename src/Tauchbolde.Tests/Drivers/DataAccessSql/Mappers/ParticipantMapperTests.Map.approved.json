{
    "dataObject": {
        "EventId": "c2847d0c-cafa-4ed8-9496-10039aa6a6f5",
        "Event": null,
        "ParticipatingDiverId": "1549f94e-e376-4ac3-9591-c6938a1c9421",
        "ParticipatingDiver": null,
        "CountPeople": 2,
        "Note": "The Note",
        "Status": 3,
        "BuddyTeamName": "Team 1",
        "Id": "bb8e0fce-974a-4877-80a9-5e4ca315e166"
    },
    "domainObject": {
        "EventId": "c2847d0c-cafa-4ed8-9496-10039aa6a6f5",
        "ParticipatingDiverId": "1549f94e-e376-4ac3-9591-c6938a1c9421",
        "ParticipatingDiver": null,
        "CountPeople": 2,
        "Note": "The Note",
        "Status": 3,
        "BuddyTeamName": "Team 1",
        "Id": "bb8e0fce-974a-4877-80a9-5e4ca315e166"
    },
    "newDataObject": {
        "EventId": "c2847d0c-cafa-4ed8-9496-10039aa6a6f5",
        "Event": null,
        "ParticipatingDiverId": "1549f94e-e376-4ac3-9591-c6938a1c9421",
        "ParticipatingDiver": null,
        "CountPeople": 2,
        "Note": "The Note",
        "Status": 3,
        "BuddyTeamName": "Team 1",
        "Id": "bb8e0fce-974a-4877-80a9-5e4ca315e166"
    }
}